enhancements/ingress/optional-ingress.md

```diff
@@ -0,0 +1,247 @@
+---
+title: optional-ingress
+authors:
+  - "@alebedev87"
+reviewers:
+  - "@Miciah"
+  - "@bparees"
+  - "@cewong"
+  - "@wking"
+approvers:
+  - "@Miciah"
+  - "@bparees"
+  - "@wking"
+api-approvers:
+  - N/A
+creation-date: 2023-06-02
+last-updated: 2023-06-02
+tracking-link:
+  - https://issues.redhat.com/browse/NE-1129
+see-also:
+  - "/enhancements/installer/component-selection.md"
+  - "/enhancements/ingress/aws-load-balancer-operator.md"
+replaces:
+superseded-by:
+---
+
+# Make Ingress Operator optional
+
+## Summary
+
+This enhancement describes the implementation of a new capability to enable/disable the default ingress of the OpenShift cluster.
+The new capability (from here on called the ingress capability) includes [the cluster ingress operator](https://github.com/openshift/cluster-ingress-operator) as well as the custom resources managed by the operator.
+This enhancement focuses on the requirements for [RFE-3395](https://issues.redhat.com/browse/RFE-3395) which targets the ROSA clusters spawned and managed by HyperShift. The implementation of the ingress capability on the standalone OpenShift is out of the scope of this enhancement.
+
+## Motivation
+
+Users of the ROSA managed OpenShift service would like to use [AWS load balancers](/enhancements/ingress/aws-load-balancer-operator.md) as a way to accept the ingress traffic into the cluster.
+The cluster ingress operator becomes unnecessary and can be disabled to:
+- reduce the resource consumption on
+  - the HyperShift client clusters (ingress controllers)
+  - the HyperShift hosted control plane (operator)
+- make the HyperShift hosted control plane lighter, simpler therefore more manageable
+- enable the HyperShift cluster users to make their choice for the cluster ingress
+
+### Goals
+- Implement the new ingress capability.
+- Add the ingress capability to the ingress operator's payload.
+- Make the dependent cluster operators tolerate the absence of the ingress operator.
+
+### Non-Goals
+- Use a different way for enabling/disabling the ingress other than via [the cluster capability](/enhancements/installer/component-selection.md).
+- Fully implement the ingress capability for the standalone OpenShift:
+  - Make the OpenShift installer tolerate the absence of the default ingress controller.
+  - Make the cluster authentication operator tolerate the absence of the default ingress controller.
+  - Make the cluster monitoring operator tolerate the absence of the default ingress controller.
+- Disable the Route API or try to tolerate the absence of the default ingress controller.
+- Implement the ingress operator disabling in the HyperShift's control plane operator.
+
+## Proposal
+
+### User Stories
```

Comment:
These are great, but I think of them as more of functional requirements. Can we have some end user focused user story? how is this providing value for them? how is this added value only desired in ROSA?

Reply:
> Can we have some end user focused user story? how is this providing value for them?

There is not much the end user would benefit from this enhancement. I can only think of the unnecessary resource consumption from the default ingress controller while the AWS LBs are in use. The rest is more in the interest of the HyperShift engineers.

> How is this added value only desired in ROSA?

It's the customer request driven. Currently ROSA needs this more than the standalone OpenShift.